host/client_integration_test.go

```diff
@@ -0,0 +1,372 @@
+	var result1 string
+	ctx1 := workflow.WithDataConverter(ctx, newTestDataConverter())
+	ctx1 = workflow.WithTaskList(ctx1, activityTaskList)
+	err1 := workflow.ExecuteActivity(ctx1, testActivity, "world1").Get(ctx1, &result1)
+	if err1 != nil {
+		return "", err1
+	}
+	return result + "," + result1, nil
+}
+
+func (s *clientIntegrationSuite) startWorkerWithDataConverter(tl string) cworker.Worker {
+	opts := cworker.Options{
+		DataConverter: newTestDataConverter(),
+	}
+	worker := cworker.New(s.wfService, s.domainName, tl, opts)
+	if err := worker.Start(); err != nil {
+		s.logger.Fatalf("Error when start worker with data converter: %v", err)
+	}
+	return worker
+}
+
+func (s *clientIntegrationSuite) TestClientDataConverter() {
+	workflow.Register(testDataConverterWorkflow)
+	activity.Register(testActivity)
+
+	worker := s.startWorkerWithDataConverter(activityTaskList)
+	defer worker.Stop()
+
+	id := "client-integration-data-converter-workflow"
+	workflowOptions := client.StartWorkflowOptions{
+		ID:                           id,
+		TaskList:                     s.taskList,
+		ExecutionStartToCloseTimeout: 20 * time.Second,
+	}
+	ctx, _ := context.WithTimeout(context.Background(), 30*time.Second)
+	we, err := s.wfClient.ExecuteWorkflow(ctx, workflowOptions, testDataConverterWorkflow)
+	if err != nil {
+		s.logger.Fatalf("Start workflow with err: %v", err)
+	}
+	s.NotNil(we)
+	s.True(we.GetRunID() != "")
+
+	var res string
+	err = we.Get(ctx, &res)
+	s.NoError(err)
+	s.Equal("hello_world,hello_world1", res)
+}
+
+var childTaskList = "client-integration-data-converter-child-tasklist"
+
+func testParentWorkflow(ctx workflow.Context) (string, error) {
+	logger := workflow.GetLogger(ctx)
+	execution := workflow.GetInfo(ctx).WorkflowExecution
+	childID := fmt.Sprintf("child_workflow:%v", execution.RunID)
+	cwo := workflow.ChildWorkflowOptions{
+		WorkflowID:                   childID,
+		ExecutionStartToCloseTimeout: time.Minute,
+	}
+	ctx = workflow.WithChildOptions(ctx, cwo)
+	var result string
+	err := workflow.ExecuteChildWorkflow(ctx, testChildWorkflow, 0, 3).Get(ctx, &result)
+	if err != nil {
+		logger.Error("Parent execution received child execution failure.", zap.Error(err))
+		return "", err
+	}
+
+	childID1 := fmt.Sprintf("child_workflow1:%v", execution.RunID)
+	cwo1 := workflow.ChildWorkflowOptions{
+		WorkflowID:                   childID1,
+		ExecutionStartToCloseTimeout: time.Minute,
+		TaskList:                     childTaskList,
+	}
+	ctx1 := workflow.WithChildOptions(ctx, cwo1)
+	ctx1 = workflow.WithDataConverter(ctx1, newTestDataConverter())
+	var result1 string
+	err1 := workflow.ExecuteChildWorkflow(ctx1, testChildWorkflow, 0, 3).Get(ctx1, &result1)
+	if err1 != nil {
+		logger.Error("Parent execution received child execution 1 failure.", zap.Error(err1))
+		return "", err1
+	}
+
+	logger.Info("Parent execution completed.", zap.String("Result", result+result1))
```

Comment:
could you assert the result and result1 here?

Reply:
added